sbi/analysis/plot.py

```diff
@@ -772,59 +1027,181 @@ def pairplot(
         )
         return fig, axes
 
-    samples, dim, limits = prepare_for_plot(samples, limits, points)
-
     # prepate figure kwargs
-    fig_kwargs_filled = _get_default_fig_kwargs()
-    # update the defaults dictionary with user provided values
-    fig_kwargs_filled = _update(fig_kwargs_filled, fig_kwargs)
+    fig_kwargs_default = FigKwargs()  # Get defaults
+    fig_kwargs_user = fig_kwargs if fig_kwargs else fig_kwargs_default
+    fig_kwargs_filled = _update(dict(fig_kwargs_default), dict(fig_kwargs_user))
+
+    # Prepare Diag Defaults
+    if type(diag) is list:
+        diag = diag[0]
+        warn(
+            f"diag uses only the first entry in a list for plot type. "
+            f"diag type is {diag}.",
+            stacklevel=2,
+        )
+
+    diag_defaults = _get_defaults("diag", plot_type=diag)
+
+    if type(diag_kwargs) is list:
+        diag_kwargs = diag_kwargs[0]
+        warn(
+            "diag_kwargs lists are currently not supported. "
+            "only the first entry is used for kwargs.",
+            stacklevel=2,
+        )
+
+    # warn the user if they provided kwargs that are ignored downstream
+    if diag_kwargs:
+        non_default_diag_kwargs = (
+            set(dict(diag_kwargs).keys())  # type: ignore
+            - set(diag_defaults.__annotations__.keys())
+        )
+        if non_default_diag_kwargs:
+            warn(
+                f"diag_kwargs has {len(non_default_diag_kwargs)} args that are "
+                "ignored by pairplot: {non_default_diag_kwargs}. " + "To pass them to "
+                "matplotlib, include them in kwargs={'mpl_kwargs':{}}."
+                "See sbi.analysis.plot.GenericMplKwargs for details.",
+                stacklevel=2,
+            )
+
+    if type(upper) is list:
+        upper = upper[0]
+        warn(
+            "upper uses only the first entry in a list for plot type. "
+            f"upper type is {upper}.",
+            stacklevel=2,
+        )
+    upper_defaults = _get_defaults("upper", plot_type=upper)
+
+    if type(upper_kwargs) is list:
+        upper_kwargs = upper_kwargs[0]
+        warn(
+            "upper_kwargs is a list. Lists are currently not supported, "
+            "only the first entry is used for kwargs.",
+            stacklevel=2,
+        )
+
+    if upper_kwargs:
+        non_default_diag_kwargs = (
+            set(dict(upper_kwargs).keys())  # type: ignore
+            - set(upper_defaults.__annotations__.keys())
+        )
+        if non_default_diag_kwargs:
+            warn(
+                f"upper_kwargs has {len(non_default_diag_kwargs)} args that are "
+                "ignored by pairplot: {non_default_diag_kwargs}. To pass them to "
```

Comment:
```suggestion
                f"ignored by pairplot: {non_default_diag_kwargs}. To pass them to "
```

Reply:
No issue is open regarding the kwargs list to my knowledge. I was going to open an issue on it after finishing this pull request but I can also do it sooner.

Also that's a great catch on the missing `f` in front of the string!